src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -545,6 +551,10 @@ public function fill(array $attributes)
             }
         }
 
+        if (! $wasUnguarded) {
```

Comment:
What's the reasoning behind checking this at this level?

Reply:
To preserve any previous "guarded" state. There's a test case that communicates this and is inline with the existing `unguard` method behavior.

Reply:
@jasonmccreary i feel try-finally would be better here, if one of the called methods throws we end up with a model that's suddenly unguarded

Reply:
Yes, that would behave more like the `guarded` callback. I'll probably make that change.